osu.Server.Queues.ScorePump/ImportHighScores.cs

```diff
@@ -23,42 +24,67 @@ namespace osu.Server.Queues.ScorePump
     [Command("import-high-scores", Description = "Imports high scores from the osu_scores_high tables into the new solo scores table.")]
     public class ImportHighScores : ScorePump
     {
+        private long lastCommitTimestamp;
+        private int currentTransactionInsertCount;
+
         [Option(CommandOptionType.SingleValue)]
         public int RulesetId { get; set; }
 
         [Option(CommandOptionType.SingleValue)]
         public long StartId { get; set; }
 
+        private const int scores_per_query = 10000;
+
+        private const int seconds_between_transactions = 2;
+
         public int OnExecute(CancellationToken cancellationToken)
         {
+            Ruleset ruleset = LegacyRulesetHelper.GetRulesetFromLegacyId(RulesetId);
+            string highScoreTable = LegacyDatabaseHelper.GetRulesetSpecifics(RulesetId).HighScoreTable;
+
             using (var dbMainQuery = Queue.GetDatabaseConnection())
             using (var db = Queue.GetDatabaseConnection())
             {
-                Ruleset ruleset = LegacyRulesetHelper.GetRulesetFromLegacyId(RulesetId);
-                string highScoreTable = LegacyDatabaseHelper.GetRulesetSpecifics(RulesetId).HighScoreTable;
-                string query = $"SELECT * FROM {highScoreTable} WHERE score_id >= @startId";
+                var transaction = db.BeginTransaction();
+
+                var insertCommand = db.CreateCommand();
 
-                Console.WriteLine($"Querying with \"{query}\"");
+                insertCommand.CommandText =
+                    // main score insert
+                    $"INSERT INTO {SoloScore.TABLE_NAME} (user_id, beatmap_id, ruleset_id, data, preserve, created_at, updated_at) "
+                    + $"VALUES (@userId, @beatmapId, {RulesetId}, @data, 0, @date, @date);"
```

Comment:
Sure about `preserve=0`? Is this because it has no function for the time being?

Reply:
That's a mistake, should be `1`